samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```diff
@@ -735,6 +741,209 @@ private MapConfig getConfigFromCoordinatorStream(Config config) {
+    // Setup input topics.
+    String statefulInputKafkaTopic = String.format("test-input-topic-%s", UUID.randomUUID().toString());
+    TestUtils.createTopic(zkUtils(), statefulInputKafkaTopic, 32, 1, servers(), new Properties());
+
+    // Generate configuration for the test.
+    Map<String, String> configMap = buildStreamApplicationConfigMap(ImmutableList.of(statefulInputKafkaTopic), testStreamAppName, testStreamAppId);
+    configMap.put(JobConfig.PROCESSOR_ID(), PROCESSOR_IDS[0]);
+    Config applicationConfig1 = new ApplicationConfig(new MapConfig(configMap));
+
+    publishKafkaEvents(statefulInputKafkaTopic, 0, NUM_KAFKA_EVENTS, PROCESSOR_IDS[0]);
+
+    // Create StreamApplication from configuration.
+    CountDownLatch kafkaEventsConsumedLatch1 = new CountDownLatch(NUM_KAFKA_EVENTS);
+    CountDownLatch processedMessagesLatch1 = new CountDownLatch(1);
+
+    ApplicationRunner appRunner1 = ApplicationRunners.getApplicationRunner(TestStreamApplication.getInstance(
+            TEST_SYSTEM, ImmutableList.of(statefulInputKafkaTopic), outputKafkaTopic, processedMessagesLatch1, null, kafkaEventsConsumedLatch1,
+            applicationConfig1), applicationConfig1);
+
+    executeRun(appRunner1, applicationConfig1);
+    processedMessagesLatch1.await();
+
+    // Generate the correct task assignments before the input stream expansion.
+    Map<TaskName, Set<SystemStreamPartition>> expectedTaskAssignments = new HashMap<>();
+    for (int partition = 0; partition < 32; ++partition) {
+      TaskName taskName = new TaskName(String.format("Partition %d", partition));
+      SystemStreamPartition systemStreamPartition = new SystemStreamPartition(TEST_SYSTEM, statefulInputKafkaTopic, new Partition(partition));
+      expectedTaskAssignments.put(taskName, ImmutableSet.of(systemStreamPartition));
+    }
+
+    // Read the latest JobModel for validation.
+    String jobModelVersion = zkUtils.getJobModelVersion();
+    JobModel jobModel = zkUtils.getJobModel(jobModelVersion);
+    Set<SystemStreamPartition> ssps = getSystemStreamPartitions(jobModel);
+
+    // Validate that the input partition count is 32 in the JobModel.
+    Assert.assertEquals(32, ssps.size());
+
+    // Validate that the new JobModel has the expected task assignments before the input stream expansion.
+    Map<TaskName, Set<SystemStreamPartition>> actualTaskAssignments = getTaskAssignments(jobModel);
+    Assert.assertEquals(expectedTaskAssignments, actualTaskAssignments);
+
+    // Increase the partition count of the input kafka topic to 64.
+    AdminUtils.addPartitions(zkUtils(), statefulInputKafkaTopic, 64, "", true, RackAwareMode.Enforced$.MODULE$);
+
+    // Wait for the JobModel version to change due to the increase in the input partition count.
+    long jobModelWaitTimeInMillis = 10;
+    while (Objects.equals(zkUtils.getJobModelVersion(), jobModelVersion)) {
```

Comment:
this may hang forever if there is an error. Add a timeout.

Reply:
We've a Junit timeout rule defined in the beginning  of this Test class that will ensures that any test execution should not take more than 60 seconds to complete.